src/plugins/intel_gpu/tests/test_cases/cl_mem_input_test.cpp

```diff
@@ -123,15 +124,33 @@ TEST(cl_mem_check, check_2_inputs) {
     topology.add(input2);
     topology.add(reorder("reorder", "input", "input2", output_layout));
 
-    network network(*engine, topology);
-    network.set_input_data("input", input_memory);
-    network.set_input_data("input2", input_memory2);
+    cldnn::network::ptr network;
 
-    auto outputs = network.execute();
+    if (is_caching_test()) {
+        membuf mem_buf;
+        {
+            cldnn::network _network(*engine, topology);
+            std::ostream out_mem(&mem_buf);
+            BinaryOutputBuffer ob = BinaryOutputBuffer(out_mem);
+            _network.save(ob);
+        }
+        {
+            std::istream in_mem(&mem_buf);
+            BinaryInputBuffer ib = BinaryInputBuffer(in_mem, *engine);
+            network = std::make_shared<cldnn::network>(ib, get_test_stream_ptr(), *engine);
+        }
+    } else {
+        network = std::make_shared<cldnn::network>(*engine, topology);
```

Comment:
[random spot] I'd expect some tests for blob size too to ensure that we don't save anything redundant.
Also, we need tests for deserialized network capabilities, as I'd expect that some functionality may be unavailable. Does perf counter/executable graph/other network API method work as expected for that scenario -- it must be checked by unit tests (can be added in separate PR)

Reply:
As you said, some functionalities are not available when deserialized. I'll add unit tests to check them in a separate PR. Thank you.